### DISCUSSION TITLE
Support for MAC address in place of device name in BluetoothA2DPSource's start()/start_raw() functions?

### DISCUSSION THREAD

**Discussion by Spartan1776** [pschatzmann/ESP32-A2DP#567] [Q&A] [answered]
Hey @pschatzmann,

First off, just wanted to say thanks a million! I'm developing my own Bluetooth MP3 (okay, really PCM haha...) player using a ESP32 Pico D4; there's not a lot out there in the way of such a venture, so when I found this project I was overjoyed.

I just tested out the BluetoothA2DPSource class and can successfully connect to my slave devices and send tones to them. In doing so, I had a question, though––would you consider adding support to allow the user to supply the Bluetooth device's MAC address in the start() and start_raw() functions in lieu of the device name? Things can get tricky with the name if: 1) there are multiple devices with the same name, or 2) the device name is empty.

I'm happy to help with this––I've been reviewing your source code, and while such a feature would require a fair number of changes, I think it can be implemented. In any case, just thought I'd throw the idea out there to get thoughts from a group of people much smarter than I.

Thanks again!

**Comment by pschatzmann** [comment, marked as answer]
Just have a look at the documentation: this scenario is already supported with the [help of a callback ](https://pschatzmann.github.io/ESP32-A2DP/html/class_bluetooth_a2_d_p_source.html#a9605c01de93792d11295125de0a24087)

Here is the [corresponding example](https://github.com/pschatzmann/ESP32-A2DP/blob/c56b11ee0ca56bc9ff2dd0bb2f4b8972e3538f02/examples/bt_music_sender_ssid_callback/bt_music_sender_ssid_callback.ino#L53)!

**Reply by Spartan1776** [reply]
I'm embarrassed I missed this, but it's perfect––thanks @pschatzmann!
